compiler/rustc_type_ir/src/ty_kind.rs

```diff
@@ -2,12 +2,13 @@ use std::fmt;
 use std::ops::Deref;
 
 use derive_where::derive_where;
+// We *would* normally import these through `rustc_data_structures`, but external
+// users (i.e. rust-analyzer) may need the impls for these traits.
+use ena::unify::{NoError, UnifyKey, UnifyValue};
```

Comment:
please add the ena import in `rustc_type_ir::data_structures` and import from there